### PR TITLE
Properly handle callback suggestion

### PR DESCRIPTION
**Pull request by msaraiva**
Properly handle callback suggestions (with or without `def/defmacro` before the cursor). 

Requires https://github.com/elixir-lsp/elixir_sense/pull/99, otherwise, suggestions will not appear after typing `"d"`, `"de"` or `"def"`.

### CODE REVIEW

**Comment by axelson** [comment]
Is it possible to add a test for this?

File: apps/language_server/test/server_test.exs [hunk 1]
```diff
@@ -265,7 +265,8 @@ defmodule ElixirLS.LanguageServer.ServerTest do
                            "severity" => 1
                          }
                        ]
-                     })
+                     }),
+                     300
```
Comment (msaraiva) [inline apps/language_server/test/server_test.exs:269]: Without that, I get timeouts all the time.

**Comment by msaraiva** [comment]
@axelson done. Testing `elixir-ls` has been a poor dev experience for me. Lots of timeouts, crashes and some times it just gets stuck and I need to run it again until all tests pass. It might be related to my environment but anyway, I'll probably start running only the tests directly related to the changes. If something goes wrong in the CI server, I'll tackle it later.

**Comment by msaraiva** [comment]
Not sure why it breaks in the CI server. Locally, a get all green:

image

As I told you, I have a very bad dev experience running the tests. Some tests run fine on my machine but not on the CI server, other tests run on the CI server but not on my machine :(

**Comment by axelson** [comment]
> @axelson done. Testing `elixir-ls` has been a poor dev experience for me. Lots of timeouts, crashes and some times it just gets stuck and I need to run it again until all tests pass. It might be related to my environment but anyway, I'll probably start running only the tests directly related to the changes. If something goes wrong in the CI server, I'll tackle it later.

Thanks for adding some tests! I am totally with you on that the current state of the test suite is quite poor. Unfortunately I too get test failures locally that I haven't been able to track down, and the tests are quite noisy (lots of output to stdout). Some of the tests are a bit flaky on CI as well. Although I did just submit #270 which cleans up the test output quite a bit.

That particular test looks to be failing because this PR doesn't actually update elixir_sense to get https://github.com/elixir-lsp/elixir_sense/pull/99

I have some changes locally that I'll push up to this branch that will make the tests not depend on the standard library and update elixir_sense.

**Comment by msaraiva** [comment]
Fantastic! You're my hero :)

Feel free to merge it whenever you're done with your changes.

Cheers.

**Comment by axelson** [comment]
Haha, thanks! I must say that it is nice to now have the test suite more under control. With #270 and #271 the test suite is completely "clean" and works reliably locally for me :confetti_ball:

**Pull request merged**